### ISSUE TITLE
Support for Netconf over TLS

### ISSUE THREAD

**Issue by danieluribe** [clicon/clixon#80]
Hello,

Is there any intention of implementing Netconf over TLS?

**Comment by olofhagsand** [comment]
Hello @danieluribe 
That would be RFC 7589? It seems a reasonable functionality to have in clixon,...
I dont know how much work it would be, it seems most about certificate binding.
Do you know of some lib or framework that would assist in the implementation?
Also, I would be interested in how you see its need - your use-case?
--Olof

**Comment by danieluribe** [comment]
Yes, I am referring to RFC 7589. I am aware that OpenSSl provides some libraries for server and client implementation for SSL3 and TLS, and also some libraries for general cryptographic and X.509 support. I don't know how hard would it be to integrate the libraries in Clixon, but it looks like a great approach.
Right now I don't have an special use case for Netconf over TLS, in my opinion with SSH is enough, but I've found some Client implementations that require the use of TLS.

**Comment by olofhagsand** [comment]
Clixon 4.6 will have a proper OpenSSL 1.1 integration using libevhtp. The step to use this for Netconf is now smaller, if of interest.
